scipy/stats/tests/test_distributions.py

```diff
@@ -3185,98 +3185,110 @@ def test_stats_broadcast(self):
-    #   A = C * (mp.digamma(C) - mp.digamma(v / 2))
-    #   B = 0.5 * mp.log(v) + mp.log(mp.beta(v / 2, mp.one / 2))
-    #   h = A + B
-    #   return float(h)
-    assert_allclose(stats.t.entropy(v), ref, rtol=1e-14)
-
-
-@pytest.mark.parametrize("methname", ["pdf", "logpdf", "cdf",
-                                      "ppf", "sf", "isf"])
-@pytest.mark.parametrize("df_infmask", [[0, 0], [1, 1], [0, 1],
-                                        [[0, 1, 0], [1, 1, 1]],
-                                        [[1, 0], [0, 1]],
-                                        [[0], [1]]])
-def test_t_inf_df(methname, df_infmask):
-    np.random.seed(0)
-    df_infmask = np.asarray(df_infmask, dtype=bool)
-    df = np.random.uniform(0, 10, size=df_infmask.shape)
-    x = np.random.randn(*df_infmask.shape)
-    df[df_infmask] = np.inf
-    t_dist = stats.t(df=df, loc=3, scale=1)
-    t_dist_ref = stats.t(df=df[~df_infmask], loc=3, scale=1)
-    norm_dist = stats.norm(loc=3, scale=1)
-    t_meth = getattr(t_dist, methname)
-    t_meth_ref = getattr(t_dist_ref, methname)
-    norm_meth = getattr(norm_dist, methname)
-    res = t_meth(x)
-    assert_equal(res[df_infmask], norm_meth(x[df_infmask]))
-    assert_equal(res[~df_infmask], t_meth_ref(x[~df_infmask]))
-
-
-@pytest.mark.parametrize("df_infmask", [[0, 0], [1, 1], [0, 1],
-                                        [[0, 1, 0], [1, 1, 1]],
-                                        [[1, 0], [0, 1]],
-                                        [[0], [1]]])
-def test_t_inf_df_stats_entropy(df_infmask):
-    np.random.seed(0)
-    df_infmask = np.asarray(df_infmask, dtype=bool)
-    df = np.random.uniform(0, 10, size=df_infmask.shape)
-    df[df_infmask] = np.inf
-    res = stats.t.stats(df=df, loc=3, scale=1, moments='mvsk')
-    res_ex_inf = stats.norm.stats(loc=3, scale=1, moments='mvsk')
-    res_ex_noinf = stats.t.stats(df=df[~df_infmask], loc=3, scale=1,
-                                 moments='mvsk')
-    for i in range(4):
-        assert_equal(res[i][df_infmask], res_ex_inf[i])
-        assert_equal(res[i][~df_infmask], res_ex_noinf[i])
-
-    res = stats.t.entropy(df=df, loc=3, scale=1)
-    res_ex_inf = stats.norm.entropy(loc=3, scale=1)
-    res_ex_noinf = stats.t.entropy(df=df[~df_infmask], loc=3, scale=1)
-    assert_equal(res[df_infmask], res_ex_inf)
-    assert_equal(res[~df_infmask], res_ex_noinf)
+    # e. g. StudentT(df=1e100).logpdf(1), StudentT(df=1e100).pdf(1)
+
+    @pytest.mark.parametrize("x, df, logpdf_ref, pdf_ref",
+                             [(1, 1e100,
+                               -1.4189385332046727, 0.24197072451914334),
+                              (1e3, 1e50, -500000.9189385332, 0.),
+                              (10, 1e20,
+                               -50.918938533204674, 7.69459862670642e-23),
+                              (1, 1,
+                               -1.8378770664093456, 0.15915494309189535)])
+    def test_logpdf_pdf(self, x, df, logpdf_ref, pdf_ref):
```

Comment:
Optional: In the past, the style has been to parameterize these tests as you have done. Parameterization is definitely preferable to adding explicit loops in tests, and it is nice if you want it to be very explicit which part of a test is failing. I'm not sure if we need either of these features in these sorts of tests, though. What do you think about defining 

```suggestion
    # caution: I haven't checked this locally for execution or PEP8
    def test_logpdf_pdf(self, x, df, logpdf_ref, pdf_ref):
        x = [1, 1e3, 10, 1]
        df = [1e100, 1e50, 1e20, 1]
        logpdf_ref = [-1.4189385332046727, -500000.9189385332,
                      -50.918938533204674, -1.8378770664093456]
        pdf_ref = [0.24197072451914334, 0, 7.69459862670642e-23, 
                   0.15915494309189535]
```
It's a little more compact and probably slightly faster to run, but the main advantage in my opinion is easy copy-paste into a script to test locally.

What do you think?

Reply:
I always like the output of `pytest -v` with parametrized tests but that's my personal preference.

What I see typically in new tests of these types of functionality in scipy are usually parametrized tests. Not sure if there are guidelines about it in the docs but we should have some level of uniformity in my opinion. So I would prefer to leave the tests as they are here.